addons/base_import/models/base_import.py

```diff
@@ -1459,9 +1464,39 @@ def execute_import(self, fields, columns, options, dryrun=False):
         # convert load's internal nextrow to the imported file's
         if import_result['nextrow']: # don't update if nextrow = 0 (= no nextrow)
             import_result['nextrow'] += skip
+        if additional_filenames:
+            import_result['additional_filenames'] = additional_filenames
 
         return import_result
 
+    def _extract_additional_filenames(self, import_fields, data, model=False, prefix='', additional_filenames=False):
+        model = model or self.res_model
+        additional_filenames = additional_filenames or {}
+        all_fields = self.env[model].fields_get()
+        for name, field in all_fields.items():
+            name = prefix + name
+            if any(name + '/' in import_field and name == import_field.split('/')[prefix.count('/')] for import_field in import_fields):
+                # Recursive call with the relational as new model and add the field name to the prefix
+                data, additional_filenames = self._extract_additional_filenames(import_fields, data, field['relation'], name + '/', additional_filenames)
+            elif field['type'] == 'binary' and field.get('attachment') and any(f in name for f in IMAGE_FIELDS) and name in import_fields:
+                additional_filenames[name] = []
+                index = import_fields.index(name)
+                for line in data:
+                    filename = None
+                    value = line[index]
+                    if isinstance(value, str):
+                        if re.match(config.get("import_image_regex", DEFAULT_IMAGE_REGEX), value):
+                            pass
+                        elif value.find('.') >= 0:
+                            # Detect if it's a filename
+                            filename = value
+                            line[index] = ''
+                        else:
+                            # base64
+                            pass
+                    additional_filenames[name].append(filename)
+        return data, additional_filenames
```

Comment:
Weird API, AFAIK, you don't change the `data` reference then why returning it ?

Reply:
`_parse_import_data_recursive` also get data as args an return data. So i have done the same.

Reply:
I understand. IMO, you should remove here. The code is very confusing with it and complex for no reason.

It will be a good idea to refactor the `_parse_import_data_recursive` one day.